File: Assets/Bundles/Shared/Resources/TS/VoxelWorld/SetupWorld.lua
-- Compiled with unity-ts v2.1.0-75
local VoxelDataAPI = require("Shared/TS/VoxelWorld/VoxelData/VoxelDataAPI").VoxelDataAPI
local EntityItemManager = require("Shared/TS/Item/HeldItems/EntityItemManager").EntityItemManager
local PrefabBlockManager = require("Shared/TS/VoxelWorld/PrefabBlockManager/PrefabBlockManager").PrefabBlockManager
local World = require("Shared/TS/VoxelWorld/World").World
local WorldAPI = require("Shared/TS/VoxelWorld/WorldAPI").WorldAPI
local function SetupWorld()
	if RunCore:IsClient() then
		local blockDefines = AssetBridge:LoadAsset("Shared/Resources/VoxelWorld/BlockDefines.xml")
		WorldAPI:GetMainWorld():LoadEmptyWorld(blockDefines, World.SKYBOX)
	end
	-- Setup Managers
	VoxelDataAPI:Init()
	PrefabBlockManager:Get()
	EntityItemManager:Get()
	-- const world = WorldService.GetWorld();
	-- const registry = world.BlockRegistry;
	-- let blockCounter = 0;
	-- for (const itemType of Object.values(ItemType)) {
	-- const itemMeta = GetItemMeta(itemType);
	-- if (!itemMeta.Block) {
	-- continue;
	-- }
	-- // print("Adding block " + itemMeta.ID + " (" + itemMeta.ItemType + ")");
	-- registry.AddBlock(
	-- BlockMeta.Make(
	-- itemMeta.ID,
	-- itemMeta.Block.TextureSide,
	-- itemMeta.Block.TextureTop ?? itemMeta.Block.TextureTopBottom ?? itemMeta.Block.TextureSide,
	-- itemMeta.Block.TextureBottom ?? itemMeta.Block.TextureTopBottom ?? itemMeta.Block.TextureSide,
	-- "cube",
	-- "",
	-- ),
	-- );
	-- blockCounter++;
	-- }
	-- print(`Registered ${blockCounter} blocks.`);
	-- WorldService.GetWorld().SetReadyToRender(1);
end
return {
	SetupWorld = SetupWorld,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Shared/VoxelWorld/SetupWorld.ts
import { BlockDataAPI } from "Shared/VoxelWorld/BlockData/BlockDataAPI";
import { EntityItemManager } from "../Item/HeldItems/EntityItemManager";
import { ItemUtil } from "../Item/ItemUtil";
import { PrefabBlockManager } from "./PrefabBlockManager/PrefabBlockManager";
import { World } from "./World";
import { WorldAPI } from "./WorldAPI";

export function SetupWorld() {
	if (RunCore.IsClient()) {
		const blockDefines = AssetBridge.LoadAsset<TextAsset>("Shared/Resources/VoxelWorld/BlockDefines.xml");
		WorldAPI.GetMainWorld().LoadEmptyWorld(blockDefines, World.SKYBOX);
	}

	// Setup Managers
	BlockDataAPI.Init();
	PrefabBlockManager.Get();
	EntityItemManager.Get();
	ItemUtil.Initialize();

	// const world = WorldService.GetWorld();
	// const registry = world.BlockRegistry;

	// let blockCounter = 0;
	// for (const itemType of Object.values(ItemType)) {
	// 	const itemMeta = GetItemMeta(itemType);
	// 	if (!itemMeta.Block) {
	// 		continue;
	// 	}

	// 	// print("Adding block " + itemMeta.ID + " (" + itemMeta.ItemType + ")");
	// 	registry.AddBlock(
	// 		BlockMeta.Make(
	// 			itemMeta.ID,
	// 			itemMeta.Block.TextureSide,
	// 			itemMeta.Block.TextureTop ?? itemMeta.Block.TextureTopBottom ?? itemMeta.Block.TextureSide,
	// 			itemMeta.Block.TextureBottom ?? itemMeta.Block.TextureTopBottom ?? itemMeta.Block.TextureSide,
	// 			"cube",
	// 			"",
	// 		),
	// 	);
	// 	blockCounter++;
	// }
	// print(`Registered ${blockCounter} blocks.`);
	// WorldService.GetWorld().SetReadyToRender(1);
}
